litex/soc/software/bios/cmds/cmd_bios.c

```diff
@@ -56,6 +107,56 @@ static void ident_handler(int nb_params, char **params)
 
 define_command(ident, ident_handler, "Identifier of the system", SYSTEM_CMDS);
 
+static void delay_time_handler(int nb_params, char **params)
+{
+	char *c = NULL;
+	volatile uint32_t start =0;
+	volatile uint32_t end = 0;
+	volatile uint32_t delay_counts = 0;
+	volatile uint32_t tick_counts = 0;
+
+
+	if (nb_params < 1) {
+		printf("delay_time <value>");
+		return;
+	}
+
+	delay_counts = strtoul(params[0], &c, 0);
+	if (*c != 0) {
+		printf("Incorrect value");
+		return;
+	}
+
+	timer0_en_write(0);
+	timer0_reload_write(0);
+	timer0_load_write(0xffffffff);
+	timer0_en_write(1);
+
+	timer0_update_value_write(1);
+	start = timer0_value_read();
+
+	delay(delay_counts);
+
+	timer0_update_value_write(1);
+	end = timer0_value_read();
+
+	tick_counts = start - end;
+	ticks_per_delay = tick_counts / delay_counts;
+	ticks_per_us = CONFIG_CLOCK_FREQUENCY / 1000000;
+
+	printf("CONFIG_CLOCK_FREQUENCY: %u\r\n", CONFIG_CLOCK_FREQUENCY);
+	printf("delay time: %u\r\n", delay_counts);
+	printf("start: %u\r\n", start);
+	printf("end:   %u\r\n", end);
+	printf("elapsed ticks: %u\r\n", tick_counts);
+	printf("ticks per delay: %u\r\n", ticks_per_delay);
+	printf("ticks per microsecond: %u\r\n", ticks_per_us);
+}
+
+define_command(delay_time, delay_time_handler, "delay time of the system", SYSTEM_CMDS);
+
+
```

Comment:
This could be  a nice example of use of the timer, but I'm still not sure it's a useful feature in the BIOS, I'll think about it.